src/pages/Admin/Details/ValidationSchema.ts

```diff
@@ -40,6 +44,37 @@ export interface RoomSchemaShape {
   image_url: string;
 }
 
+const venueNameLengthValidation = Yup.string()
+  .required("Venue name is required!")
+  .min(
+    VENUE_NAME_MIN_CHAR_COUNT,
+    ({ min }) => `Name must be at least ${min} characters`
+  )
+  .max(
+    VENUE_NAME_MAX_CHAR_COUNT,
+    ({ max }) => `Name must be less than ${max} characters`
+  );
+
+const venueNameValidation = venueNameLengthValidation
+  .test(
+    "name",
+    "Must have alphanumeric characters",
+    (val: string) => createUrlSafeName(val).length > 0
+  )
+  .test(
+    "name",
+    "This venue name is already taken",
+    async (val: string) =>
+      !val ||
+      !(
+        await firebase
+          .firestore()
+          .collection("venues")
+          .doc(createUrlSafeName(val))
+          .get()
+      ).exists
+  );
```

Comment:
How often is this validation going to get executed? I'm guessing not just when the form is submitted? So potentially on every keystroke/similar? This might be a bit heavyweight to include here, and would likely be better done as a backend-triggered error anyway, since we're going to have to check it there regardless.

Reply:
It depends on the UI. `useForm` accepts `reValidateMode` parameter which can be `onChange`, `onSubmit` and `onBlur`. I'm currently using `onSubmit` but if someone changes it to `onChange` it will be heavy indeed.

Reply:
I agree that this should be removed from here, but every single of the admin creation tools is using this and removing the validation means that I need to go and handle every single 403 status code. Maybe I can create an issue for this and we can fix it in another PR?

Reply:
Ya, backlog it for future cleanup

Reply:
Also maybe add an ‘@debt’ comment here calling out the onBlur/etc thing you mentioned above